### ISSUE TITLE
- Crash [PayUValidations validateAmount:] 

### ISSUE THREAD

**Issue by aasawa99** [payu-intrepos/iOS-SDK#15]
Sdk is crashing for few users with following crash information.

Fatal Exception: NSInvalidArgumentException
-[__NSCFNumber compare:]: nil argument

```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x18458b164 __exceptionPreprocess
1  libobjc.A.dylib                0x1837d4528 objc_exception_throw
2  CoreFoundation                 0x18458b0ac -[NSException initWithCoder:]
3  CoreFoundation                 0x184458c4c -[__NSCFNumber compare:]
4  latte                          0x1012a2628 -[PayUValidations validateAmount:] (PayUValidations.m:657)
5  latte                          0x10129fe2c -[PayUValidations validateMandatoryParamsForPaymentHashGeneration:] (PayUValidations.m:65)
6  latte                          0x10129fbbc -[PayUValidations validateMandatoryParamsForPayment:] (PayUValidations.m:41)
7  latte                          0x1012a6488 -[PayUCreateRequest createRequestWithPaymentParam:forPaymentType:withCompletionBlock:] (PayUCreateRequest.m:50)
8  latte                          0x100aeac54 DZPaymentManager.payFromCard(String, amount : String, hashString : String, cardData : DZPayUCard, cvv : String, viewCntlr : UIViewController, transactionData : DZTransactionData) -> () empty-list NSError? first-element-marker  empty-list  (DZPaymentManager.swift:877)
9  latte                          0x100ada324 DZPaymentManager.payFromCard(String, amount : String, hashString : String, cardData : DZPayUCard, cvv : String, viewCntlr : UIViewController, transactionData : DZTransactionData) -> () empty-list NSError? first-element-marker  empty-list  (DZPaymentManager.swift)
10 latte                          0x100ae9b7c DZPaymentManager.payFromCard(String, amount : String, hashString : String, cardData : DZPayUCard, cvv : String, viewCntlr : UIViewController, transactionData : DZTransactionData) -> () empty-list NSError? first-element-marker  empty-list  (DZPaymentManager.swift)
11 latte                          0x100b844cc DZPayUManagerloadMoneyRequest empty-list NSNumber first-element-marker Boolis_auto_debitNSNumber?total_debit_amountString?task_id (DZPayUManager.swift:581)
12 latte                          0x100e6f404 thunk (SearchTextField.swift)
13 libdispatch.dylib              0x183f0aa54 _dispatch_call_block_and_release
14 libdispatch.dylib              0x183f0aa14 _dispatch_client_callout
15 libdispatch.dylib              0x183f4bc80 _dispatch_main_queue_callback_4CF$VARIANT$armv81
16 CoreFoundation                 0x184533344 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
17 CoreFoundation                 0x184530f20 __CFRunLoopRun
18 CoreFoundation                 0x184450c58 CFRunLoopRunSpecific
19 GraphicsServices               0x1862fcf84 GSEventRunModal
20 UIKit                          0x18dba95c4 UIApplicationMain
21 latte                          0x1009de60c main (AppDelegate.swift:114)
22 libdyld.dylib                  0x183f7056c start
```

**Comment by AshishPayU** [comment]
Hi Aaswa,
Please make sure you are passing NSString in post params, passing NSNumber to the post params will cause the issue.

**Comment by ganpats** [comment]
I also facing the same issue, last i faced on 26-July-2019, using SDK ver. 3.9.2.
I'm sending parameter as NSString, but still problem exist.

Please fix as soon as possible.

**Comment by umangarya336** [comment]
This issue is fixed in our latest version 4.1.0
Please drop mail mobile.integration@payu.in if issue still exists.
